frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx

```diff
@@ -2,26 +2,66 @@ import { isEqual } from 'lodash'
 import create from 'zustand'
 import { devtools } from 'zustand/middleware'
 
-import { FormStartPage } from '~shared/types'
+import { CustomFormLogo, FormLogoBase, FormStartPage } from '~shared/types'
+
+import { UploadedImage } from './BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditImage/UploadImageInput'
+
+export type CustomLogoMeta = Omit<CustomFormLogo, keyof FormLogoBase>
+
+/** Design drawer form input fields. DesignStore keeps track of the data in the
+ * drawer input as a single unit. Other data (specifically the logo metadata) is
+ * kept separately with its own getters and setters.
+ */
+export type FormStartPageInput = Omit<
+  FormStartPage,
+  'logo' | 'estTimeTaken'
+> & {
+  estTimeTaken: number | ''
+  logo: FormLogoBase
+  attachment: UploadedImage // Custom logo image
+}
 
 export type DesignStore = {
-  startPageData: FormStartPage | null
-  setStartPageData: (startPage: FormStartPage) => void
-  resetStartPageData: () => void
+  startPageData?: FormStartPageInput
+  customLogoMeta?: CustomLogoMeta
+  setStartPageData: (startPageInput: FormStartPageInput) => void
+  setAttachment: (attachment: UploadedImage) => void
+  setCustomLogoMeta: (customLogoMetaData: CustomLogoMeta) => void
+  resetCustomLogoMeta: () => void
+  resetDesignStore: () => void
 }
 
 export const useDesignStore = create<DesignStore>(
   devtools((set, get) => ({
-    startPageData: null,
-    setStartPageData: (startPage) => {
+    setStartPageData: (startPageData: FormStartPageInput) => {
+      const current = get()
+      if (isEqual(current.startPageData, startPageData)) return
+      set({ startPageData })
+    },
+    setAttachment: (attachment: UploadedImage) => {
       const current = get()
-      if (isEqual(current.startPageData, startPage)) return
+      if (!current.startPageData) return
+      if (isEqual(current.startPageData?.attachment, attachment)) return
```

Comment:
/nit: since `current.startPageData` is checked above, the conditional access is not needed here:
```suggestion
      if (isEqual(current.startPageData.attachment, attachment)) return
```

Reply:
You're right! thanks for catching that